integrations/sklearn/neptune_sklearn/__init__.py

```diff
@@ -0,0 +1,1190 @@
+
+    if log_charts:
+        reg_summary['diagnostics_charts'] = {
+            'learning_curve': create_learning_curve_chart(regressor, X_train, y_train),
+            'feature_importance': create_feature_importance_chart(regressor, X_train, y_train),
+            'residuals': create_residuals_chart(regressor, X_train, X_test, y_train, y_test),
+            'prediction_error': create_prediction_error_chart(regressor, X_train, X_test, y_train, y_test),
+            'cooks_distance': create_cooks_distance_chart(regressor, X_train, y_train)}
+
+    return reg_summary
+
+
+def create_classifier_summary(classifier, X_train, X_test, y_train, y_test, nrows=1000, log_charts=True):
+    """Create sklearn classifier summary.
+
+    This method creates classifier summary that includes:
+
+    * all classifier parameters,
+    * pickled estimator (model),
+    * test predictions,
+    * test predictions probabilities,
+    * test scores,
+    * model performance visualizations.
+
+    Returned ``dict`` can be assigned to the experiment's namespace defined by the user (see example below).
+
+    Classifier should be fitted before calling this function.
+
+    Tip:
+        Check Sklearn-Neptune integration
+        `documentation <https://docs-beta.neptune.ai/essentials/integrations/machine-learning-frameworks/sklearn>`_
+        for the full example.
+
+    Args:
+        classifier (:obj:`classifier`):
+            | Fitted sklearn classifier object
+        X_train (:obj:`ndarray`):
+            | Training data matrix
+        X_test (:obj:`ndarray`):
+            | Testing data matrix
+        y_train (:obj:`ndarray`):
+            | The classification target for training
+        y_test (:obj:`ndarray`):
+            | The classification target for testing
+        nrows (`int`, optional, default is 1000):
+            | Log first ``nrows`` rows of test predictions and predictions probabilities.
+        log_charts (:bool:, optional, default is ``True``):
+            | If True, calculate and send chart visualizations.
+            |
+            | NOTE: calculating visualizations is potentially expensive depending on input data and classifier, and
+            | may take some time to finish.
+            |
+            | This is equivalent to calling ``log_classification_report_chart``, ``log_confusion_matrix_chart``,
+            | ``log_roc_auc_chart``, ``log_precision_recall_chart``, ``log_class_prediction_error_chart``
+            | functions from this module.
+
+    Returns:
+        ``dict`` with all summary items.
+
+    Examples:
+        Log random forest classifier summary.
+
+        .. code:: python3
+
+            import neptune.new.integrations.sklearn as npt_utils
+
+            rfc = RandomForestClassifier()
+            rfc.fit(X_train, y_train)
+
+            exp = neptune.init(project='my_workspace/my_project')
+            exp['summary/random_forest'] = npt_utils.create_classifier_summary(rfc, X_train, X_test, y_train, y_test)
```

Comment:
Information architecture wise shouldn't it be exp['random_forest/summary']?

Reply:
corrected.